pkg/registry/apis/provisioning/jobs/syncer.go

```diff
@@ -455,8 +455,7 @@ func (r *Syncer) ensureFolderPathExists(ctx context.Context, dirPath, parent str
 
 // ensureFolderExists creates the folder if it doesn't exist.
 // If the folder already exists:
-// - it will update the repository info.
-// - it will keep the parent folder already set
+// - it will error if the folder is not owned by this repository
```

Comment:
Rather than changing the target folder properties -- we should fail if the target is not owned by this repository.